### DISCUSSION TITLE
Accept a User object in the author field of an embed

### DISCUSSION THREAD

**Discussion by angelnext** [discordjs/discord.js#8392] [Ideas]
Currently if you want to add a user as an author you must set each property individually
```js
const embed = new EmbedBuilder({
  author: {
    name: user.tag,
    icon_url: user.avatarURL(),
  }
})
```

What I wish I could do is this
```js
const embed = new EmbedBuilder({
  author: user,
})
```
If this is not possible I would like to know why, thank you

**Comment by iCrawl** [comment]
The only thing that comes close to what you can do better than passing it in the constructor (which only accepts raw data, directly from discord) is:

```ts
const embed = new EmbedBuilder().setAuthor({ name: user.tag, iconURL: user.displayAvatarURL() });
```

We don't allow passing structures directly because it would be on us to serialize them and it's not really our responsibility. We'd have to have additional checks depending on what you pass in. Theoretically you could pass in a guild right? Since it has a name and an icon too. Where does it end? :P

**Reply by angelnext** [reply]
Thank you for such a straightforward answer, I understand it much better now.
